packages/webapp/src/_app/hooks/queries.ts

```diff
@@ -267,14 +269,32 @@ export const useIsCommunityActive = () => {
     };
 };
 
-export const useMyDelegation = (queryOptions: any = {}) => {
+/**
+ * Use this hook for querying signed-in user's delegation from a past/completed election. Pass in memberStats for retrieving fresh, during-election delegation results.
+ * @param { MemberStats } memberStats - if omitted, internal useMemberStats() query will be used; include if you want to trigger rerender using fresher data
+ * @param { any } queryOptions - any react-query queryOptions; enabled will be merged
+ * @returns query of <EdenMember[], Error>
+ */
+export const useMyDelegation = ({
+    memberStats,
+    queryOptions = {},
+}: {
+    memberStats?: MemberStats;
+    queryOptions?: any;
+}) => {
     const { data: member } = useCurrentMember();
-    // use queryOptions.enabled unless unspecified, in which case, ensure we don't disable the internal `enabled`
-    let enabled = "enabled" in queryOptions ? queryOptions.enabled : true;
+    const { data: cachedMemberStats } = useMemberStats({
+        enabled: !memberStats,
+    });
 
-    return useQuery({
-        ...queryMyDelegation(member?.account),
-        enabled: enabled && Boolean(member?.account),
+    const stats = memberStats ?? cachedMemberStats;
+    const enabled = Boolean(member?.account && stats);
+    const highestCompletedRoundIndex = stats && stats.ranks.length - 1;
```

Comment:
op to simplify:
const highestCompletedRoundIndex = stats?.ranks.length - 1

Reply:
Ah, right! The reason we can't do this here is because we follow it with an arithmetic operation. We cannot subtract `1` from `undefined`. So in this case, it _must_ be `stats && stats.ranks.length - 1`.